### DISCUSSION TITLE
[BUG] REST API updateRelease not working

### DISCUSSION THREAD

**Discussion by subnut** [community/community#24332] [Apps, API and Webhooks]
<p>The github REST API for updating a release using a release-id isn’t working</p>
<p>Example taken from <a href="https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#update-a-release" rel="noopener nofollow ugc">here</a> -</p>
<pre><code class="lang-auto">curl \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/35728483
</code></pre>
<p>returns</p>
<pre><code class="lang-auto">{
  "url": "https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/35728483",
  "assets_url": "https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/35728483/assets",
  "upload_url": "https://uploads.github.com/repos/subnut/nvim-ghost.nvim/releases/35728483/assets{?name,label}",
  "html_url": "https://github.com/subnut/nvim-ghost.nvim/releases/tag/v0.0.33",
  "id": 35728483,
  "author": {
    "login": "github-actions[bot]",
    "id": 41898282,
    "node_id": "MDM6Qm90NDE4OTgyODI=",
    "avatar_url": "https://avatars2.githubusercontent.com/in/15368?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/github-actions%5Bbot%5D",
    "html_url": "https://github.com/apps/github-actions",
    "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
    "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
    "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
    "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
    "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
    "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
    "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
    "type": "Bot",
    "site_admin": false
  },
  "node_id": "MDc6UmVsZWFzZTM1NzI4NDgz",
  "tag_name": "v0.0.33",
  "target_commitish": "c6eeeb23ffd89ee7cecddbd414f5fe8a2be02b3a",
  "name": "",
  "draft": false,
  "prerelease": true,
  "created_at": "2020-12-26T11:59:02Z",
  "published_at": "2020-12-26T11:59:57Z",
  "assets": [
    {
      "url": "https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/assets/29988850",
      "id": 29988850,
      "node_id": "MDEyOlJlbGVhc2VBc3NldDI5OTg4ODUw",
      "name": "nvim-ghost-linux.tar.gz",
      "label": "",
      "uploader": {
        "login": "github-actions[bot]",
        "id": 41898282,
        "node_id": "MDM6Qm90NDE4OTgyODI=",
        "avatar_url": "https://avatars2.githubusercontent.com/in/15368?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
        "html_url": "https://github.com/apps/github-actions",
        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
        "type": "Bot",
        "site_admin": false
      },
      "content_type": "application/zip",
      "state": "uploaded",
      "size": 15841828,
      "download_count": 0,
      "created_at": "2020-12-26T12:00:42Z",
      "updated_at": "2020-12-26T12:00:43Z",
      "browser_download_url": "https://github.com/subnut/nvim-ghost.nvim/releases/download/v0.0.33/nvim-ghost-linux.tar.gz"
    },
    {
      "url": "https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/assets/29988846",
      "id": 29988846,
      "node_id": "MDEyOlJlbGVhc2VBc3NldDI5OTg4ODQ2",
      "name": "nvim-ghost-macos.tar.gz",
      "label": "",
      "uploader": {
        "login": "github-actions[bot]",
        "id": 41898282,
        "node_id": "MDM6Qm90NDE4OTgyODI=",
        "avatar_url": "https://avatars2.githubusercontent.com/in/15368?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
        "html_url": "https://github.com/apps/github-actions",
        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
        "type": "Bot",
        "site_admin": false
      },
      "content_type": "application/zip",
      "state": "uploaded",
      "size": 7473323,
      "download_count": 0,
      "created_at": "2020-12-26T12:00:36Z",
      "updated_at": "2020-12-26T12:00:37Z",
      "browser_download_url": "https://github.com/subnut/nvim-ghost.nvim/releases/download/v0.0.33/nvim-ghost-macos.tar.gz"
    },
    {
      "url": "https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/assets/29988858",
      "id": 29988858,
      "node_id": "MDEyOlJlbGVhc2VBc3NldDI5OTg4ODU4",
      "name": "nvim-ghost-win64.zip",
      "label": "",
      "uploader": {
        "login": "github-actions[bot]",
        "id": 41898282,
        "node_id": "MDM6Qm90NDE4OTgyODI=",
        "avatar_url": "https://avatars2.githubusercontent.com/in/15368?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/github-actions%5Bbot%5D",
        "html_url": "https://github.com/apps/github-actions",
        "followers_url": "https://api.github.com/users/github-actions%5Bbot%5D/followers",
        "following_url": "https://api.github.com/users/github-actions%5Bbot%5D/following{/other_user}",
        "gists_url": "https://api.github.com/users/github-actions%5Bbot%5D/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/github-actions%5Bbot%5D/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/github-actions%5Bbot%5D/subscriptions",
        "organizations_url": "https://api.github.com/users/github-actions%5Bbot%5D/orgs",
        "repos_url": "https://api.github.com/users/github-actions%5Bbot%5D/repos",
        "events_url": "https://api.github.com/users/github-actions%5Bbot%5D/events{/privacy}",
        "received_events_url": "https://api.github.com/users/github-actions%5Bbot%5D/received_events",
        "type": "Bot",
        "site_admin": false
      },
      "content_type": "application/zip",
      "state": "uploaded",
      "size": 8104128,
      "download_count": 0,
      "created_at": "2020-12-26T12:01:30Z",
      "updated_at": "2020-12-26T12:01:31Z",
      "browser_download_url": "https://github.com/subnut/nvim-ghost.nvim/releases/download/v0.0.33/nvim-ghost-win64.zip"
    }
  ],
  "tarball_url": "https://api.github.com/repos/subnut/nvim-ghost.nvim/tarball/v0.0.33",
  "zipball_url": "https://api.github.com/repos/subnut/nvim-ghost.nvim/zipball/v0.0.33",
  "body": ""
}
</code></pre>
<p>Which is correct and indicates that the release id <code>35728483</code> is valid.</p>
<p>But then,</p>
<pre><code class="lang-auto">curl \
  -X PATCH \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/35728483 \
  -d '{"prerelease":true}
</code></pre>
<p>returns</p>
<pre><code class="lang-auto">{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/repos#update-a-release"
}
</code></pre>

**Comment by hubotim** [comment]
<p>When you have my instructions let me know</p>

**Comment by subnut** [comment]
<p>I edited the original post to include instructions, please check</p>

**Comment by parithiban** [comment]
<p><a class="mention" href="/subnut">@subnut</a></p>
<p>Are you passing your Authorization token in header while calling your request?</p>
<p>If so check wether it has necessary scopes</p>

**Comment by subnut** [comment]
<pre><code class="lang-auto">❯  curl -u "subnut" -i -X PATCH -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/subnut/nvim-ghost.nvim/releases/35728483 -d '{"prerelease":true}'
Enter host password for user 'subnut':
HTTP/1.1 404 Not Found
Date: Thu, 07 Jan 2021 13:48:21 GMT
Content-Type: application/json; charset=utf-8
Content-Length: 117
Server: GitHub.com
Status: 404 Not Found
X-GitHub-Media-Type: github.v3; format=json
X-RateLimit-Limit: 60
X-RateLimit-Remaining: 59
X-RateLimit-Reset: 1610030901
X-RateLimit-Used: 1
Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, Deprecation, Sunset
Access-Control-Allow-Origin: *
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
Vary: Accept-Encoding, Accept, X-Requested-With
X-GitHub-Request-Id: 8396:1F92:161728F:1C27A09:5FF71125

{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/repos#update-a-release"
}
</code></pre>
<p>I used the above. I entered my github password where required. Isn’t that enough?</p>

**Comment by parithiban** [comment, marked as answer]
<p>Can you try creating a <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token" rel="noopener nofollow ugc">PAT</a> token with the necessary scopes and check it.</p>

**Comment by subnut** [comment]
<p><a class="mention" href="/parithiban">@parithiban</a> Thanks! That has finally got it working.</p>
<p>In fact, I just discovered that the username-password method of authentication has been deprecated. Here -</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/rest/overview/other-authentication-methods" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/free-pro-team@latest/rest/overview/other-authentication-methods" target="_blank" rel="noopener nofollow ugc">Other authentication methods - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/free-pro-team@latest/rest/overview/other-authentication-methods" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/free-pro-team@latest/rest/overview/other-authentication-methods</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by Shahreyar46** [comment]
Hello,  I Implemented this function  to get releases. But zipball_url not working, I implement this for private repository.

private function get_repository_info() {
        
        if (is_null($this->github_response)) {
            $request_uri = sprintf('https://api.github.com/repos/%s/%s/releases', $this->username, $this->repository);

    
            // Switch to HTTP Basic Authentication for GitHub API v3
            $curl = curl_init();

            curl_setopt_array($curl, [
                CURLOPT_URL => $request_uri,
                CURLOPT_RETURNTRANSFER => true,
                CURLOPT_ENCODING => "",
                CURLOPT_MAXREDIRS => 10,
                CURLOPT_TIMEOUT => 0,
                CURLOPT_FOLLOWLOCATION => true,
                CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
                CURLOPT_CUSTOMREQUEST => "GET",
                CURLOPT_HTTPHEADER => [
                    "Accept: application/vnd.github+json",
                    "Authorization: Bearer $this->authorize_token",
                    "X-Github-Api-Version: 2022-11-28",
                    'User-Agent: Shahreyar46'
                ]
            ]);

            $response = curl_exec($curl);

            curl_close($curl);

            $response = json_decode($response, true);

            if (is_array($response)) {
                $response = current($response);
            }

            if ($this->authorize_token) {
                $response['zipball_url'] = add_query_arg('access_token', $this->authorize_token, $response['zipball_url']);
            }

            $this->github_response = $response;
        }
    }

Its works fine and return below response.
(
    [url] => https://api.github.com/repos/Shahreyar46/plugin-updater/releases/119805521
    [assets_url] => https://api.github.com/repos/Shahreyar46/plugin-updater/releases/119805521/assets
    [upload_url] => https://uploads.github.com/repos/Shahreyar46/plugin-updater/releases/119805521/assets{?name,label}
    [html_url] => https://github.com/Shahreyar46/plugin-updater/releases/tag/2.0.0
    [node_id] => RE_kwDOKPJ52M4HJBZR
    [tag_name] => 2.0.0
    [target_commitish] => master
    [name] => version 2.0.0
    [draft] => 
    [prerelease] => 
    [created_at] => 2023-09-04T11:34:00Z
    [published_at] => 2023-09-04T11:35:22Z
    [assets] => Array
        (
        )

    [tarball_url] => https://api.github.com/repos/Shahreyar46/plugin-updater/tarball/2.0.0
    [zipball_url] => https://api.github.com/repos/Shahreyar46/plugin-updater/zipball/2.0.0
    [body] => New test version
)

Get this message when try to access zipball url. Can you help me to find out the issue?
{
"message": "Not Found",
"documentation_url": "https://docs.github.com/rest/repos/contents#download-a-repository-archive-zip"
}
